lib/credit_card_widget.dart

```diff
@@ -160,43 +167,111 @@ class CreditCardWidget extends StatefulWidget {
   /// Provides border at back of credit card widget.
   final BoxBorder? backCardBorder;
 
+  /// The config for making the card float as per the movement of device or
+  /// mouse pointer.
+  final FloatConfig? floatConfig;
+
+  /// floating animation enabled/disabled
   @override
   _CreditCardWidgetState createState() => _CreditCardWidgetState();
 }
 
 class _CreditCardWidgetState extends State<CreditCardWidget>
-    with SingleTickerProviderStateMixin {
+    with SingleTickerProviderStateMixin, WidgetsBindingObserver {
   late AnimationController controller;
   late Animation<double> _frontRotation;
   late Animation<double> _backRotation;
   late Gradient backgroundGradientColor;
-  late bool isFrontVisible = true;
-  late bool isGestureUpdate = false;
 
+  bool isFrontVisible = true;
+  bool isGestureUpdate = false;
   bool isAmex = false;
 
+  late bool isFloatingAnimationEnabled = widget.floatConfig?.isEnabled ?? false;
+
+  final FloatingController floatingAnimController =
+      FloatingController.predefined(
+    isGyroscopeAvailable:
+        FlutterCreditCardPlatform.instance.isGyroscopeAvailable,
+  );
+
+  final StreamController<FloatingEvent> frontCardStreamController =
+      StreamController<FloatingEvent>.broadcast();
+
+  final StreamController<FloatingEvent> backCardStreamController =
+      StreamController<FloatingEvent>.broadcast();
+
+  Orientation? orientation;
+  Size? screenSize;
+
+  bool isAnimating = false;
+
+  double get glarePosition =>
```

Comment:
Please add some documentation above this getter so that anyone can understand what this logic does.